src/prowjobsscraper/equinix.py

```diff
@@ -0,0 +1,66 @@
+import logging
+from typing import Final, Optional
+
+from google.cloud import exceptions, storage  # type: ignore
+
+from prowjobsscraper import utils
+from prowjobsscraper.prowjob import EquinixMetadata, ProwJob, ProwJobs
+
+logger = logging.getLogger(__name__)
+
+
+class EquinixExtractor:
+    """
+    EquinixExtractor parses the Equinix metadata generated by the gather step in the assisted-baremetal workflow.
+    """
+
+    # map cloud cluster profile with the possible locations of the equinix metadata file
+    _METADATA_PATH_TEMPLATES: Final[dict[str, list[str]]] = {
+        "packet-assisted": [
+            "{}/artifacts/{}/assisted-baremetal-gather/artifacts/equinix-metadata.json",
+            "{}/artifacts/{}/assisted-baremetal-operator-gather/artifacts/equinix-metadata.json",
+        ],
+        "packet-sno": [
+            "{}/artifacts/{}/baremetalds-sno-gather/artifacts/equinix-metadata.json"
+        ],
+    }
+
+    def __init__(self, client: storage.Client):
+        self._client = client
+
+    def hydrate(self, jobs: ProwJobs) -> None:
```

Comment:
what's your intention by the verb ``hydrate``?
is it like enriching the existing data with more info?

Reply:
yes, that is the idea